### ISSUE TITLE
Multi-class segmentation

### ISSUE THREAD

**Issue by shanpriya3** [jeya-maria-jose/Medical-Transformer#43]
Hi, 
First of all, congrats on your work on Medical Transformer, and thanks for sharing the code. Currently, I am working on transformer-based segmentation tasks and I would like to use your code on my data. Is it possible to use this code for multi-class segmentation?

**Comment by jeya-maria-jose** [comment]
Yes, you can just change the num_classes argument to number of classes in https://github.com/jeya-maria-jose/Medical-Transformer/blob/c9c70648369ec5fb7b7f6240f75eeef09494558d/lib/models/axialnet.py#L511 to convert the code for multi-class segmentation.

**Comment by li-yanling** [comment]
> 

did you change your loss function to categorical cross entropy instead of binary cross entropy?

**Comment by diogocid** [comment]
First of all, thank you very much for making Medical Transformer availablet. However, I am facing some difficulties in adapting the model for segmenting 3 distinct classes.

Here's what I've done:

My masks have three different colors to represent three classes (class 1: orange (255, 165, 0), class 2: blue (0, 0, 128) and background: black (0, 0, 0)).
I've already changed the num_classes parameter in axialnet.py to 3, as mentioned in issue #43.
However, I'm having trouble ensuring that the template correctly identifies each class. I know that masks need to be mapped to numeric labels, but I'm not sure where and how to add this conversion in the pipeline.

My question is:

Where exactly should I add the logic to convert the colors of the masks (orange, blue and black) into numeric labels (0, 1, 2) in the code?
Do I need to modify anything additional in the preprocessing code to ensure that the three classes are correctly handled during training?
If anyone can guide me or share an example of how to adapt this part of the code, it would be a great help!

Thank you in advance for your attention and help.
